### PR TITLE
[357] Adds functionality to handle 404 errors

### PR DESCRIPTION
**Pull request by kushniryb**
Resolves [#357](https://github.com/pivorakmeetup/pivorak-web-app/issues/357)

### CODE REVIEW

**Comment by denys-medynskyi** [comment]
1. why no specs?
2. consider this case
img

**Comment by kushniryb** [comment]
@denysmedynskyi Hold on with this for now, seems like previously we were rendering 404 template but @sveredyuk changed it to raise an exception which should've worked considering that Rails has a built-in mechanism for that but it doesn't :man_facepalming:

**Comment by kushniryb** [comment]
@denysmedynskyi @sveredyuk 

Seems like I've embarked on fixing the wrong issue, lol :man_facepalming: 

Rails handles 400 & 500 errors by default using `ActionDispatch::ExceptionWrapper.rescue_responses` mapping so there's no need to manually render file templates which is nice.

The problem is that `Rollbar` considers `ActionController::RoutingError` as a warning and `ActiveRecord::RecordNotFound` as an error therefore we're constantly getting those notifications that something went wrong. 

As it turns out `PagesController#show` method is smart enough to check whether a page is present and if not - it just raises `ActionController::RoutingError`, so basically everything is taken care of meaning requests such as `pivorak.com/asd`, `pivorak.com/something` are handled gracefully.

I believe we can apply the same technique to `TalksController#show` action(although we'll have to use `find_by(id: params[:id]) || find_by(slug: params[:slug])` or we can just change `ActiveRecord::NotFound` level to warning via `Rollbar`. 

Basically there's no critical issue, it's just that we're being misguided by `Rollbar` :|

Let me know what you think about it

**Review by sveredyuk — DISMISSED** [review]

**Comment by sveredyuk** [comment]
I think it needs at least 1 spec :)

**Comment by kushniryb** [comment]
@sveredyuk @denysmedynskyi 
Done, take a look once again

**Review by sveredyuk — APPROVED** [review]
